### PR TITLE
Tc39 type erasure test fixes

### PR DESCRIPTION
**Pull request by mstoykov**
## What?

Fix some issues with tc39 tests

## Why?

Before tc39 tests for enhanced compatilbity mode aka type erasure for ts were basically not doing anything.

Also turns out compiler.Parse was failing some tests due to lack of support for corner cases needed only in tc39 tests.

More info in each commit message.

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`make lint`) and all checks pass.
- [x] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: js/modules/resolution.go [hunk 1]
```diff
@@ -103,24 +104,17 @@ func (mr *ModuleResolver) resolveLoaded(basePWD *url.URL, arg string, data []byt
 	if cached, ok := mr.cache[specifier.String()]; ok {
 		return cached.mod, cached.err
 	}
-	prg, _, err := mr.compiler.Parse(string(data), specifier.String(), false)
+	prg, _, err := mr.compiler.Parse(string(data), specifier.String(), false, true)
```
Comment (olegbespalov) [inline js/modules/resolution.go:107]: Shouldn't the last `true` be computed?

```suggestion
	prg, _, err := mr.compiler.Parse(string(data), specifier.String(), false, isESM(prg))
```

Comment (olegbespalov) [reply js/modules/resolution.go:107]: :facepalm:

**Review by olegbespalov — APPROVED** [review]

**Review by oleiade — APPROVED** [review]
